sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -646,6 +651,18 @@ which is used in a graph will be kept alive on the host for the lifetime of the
 graph. Destroying that buffer during the lifetime of a `command_graph`
 constructed with this property results in undefined behavior.
 
+===== Enable-Profiling Property [[enable-profiling]]
+
+The `property::graph::enable_profiling` property can be passed to the 
+`command_graph::finalize()` function and enables profiling support 
+for the returned `command_graph<graph_state::executable>`.
+Passing this property to the finalize function implies disabling 
+certain optimizations to enable graph profiling.
+As a result, the execution time of a graph finalized with profiling enabled 
+is longer than that of a graph without profiling capability.
+An error will be thrown when attempting to profile an event 
+from a graph submission that was created without this property.
```

Comment:
For Record & Replay, do we need to pass both the queue profiling property and the graph property to be able to profile graph events?

Reply:
Yes, we need both in the current implementation.

Reply:
I've updated the code (and the spec) accordingly.
Now, users must enable the profiling on at least one node to enable the profiling of the graph,
This can be done by using either the `property::graph::enable_profiling` property in building mode or the `property::queue::enable_profiling` on the recorded queue (Record&Replay API).
Note that the `property::queue::enable_profiling` must also set on the queue used to submit the graph. The event submission timestamp is indeed managed by the runtime, and is only set when submit queue has the flag. This might change with the profiling tag...

Reply:
It's not implemented yet, but I think it might be useful to add a new API to disable profiling on nodes. Such as `node::update_properties(const property_list &PropList)` with a new property `property::queue::disable_profiling`?